File: Level 2/올바른 괄호/solution.js
function solution(brackets) {
  if (brackets[0] === ")") {
    return false;
  }

  const stack = [];

  for (const bracket of brackets.split("")) {
    if (bracket === "(") {
      stack.push(bracket);
    } else {
      if (stack.length > 0) {
        stack.pop();
      } else {
        return false;
      }
    }
  }

  return stack.length > 0 ? false : true;
}

File: Level 2/올바른 괄호/solution.py
def solution(s):
    if s[0] == ")":
        return False

    stack = []

    for bracket in s:
        if bracket == "(":
            stack.append(bracket)
        else:
            if not len(stack):
                return False
            else:
                stack.pop()

    return False if stack else True
